src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeController.java

```diff
@@ -159,16 +159,28 @@ protected DataNode buildEmptyDataNode() {
     @RequestMapping(value = "/api/v1/data-nodes/{dataNodeId}", method = RequestMethod.PUT)
     public JsonResult<DataNodeDTO> updateDataNode(
             @PathVariable("dataNodeId") Long dataNodeId,
-            @RequestBody @Valid CommonDataNodeRegisterDTO commonDataNodeRegisterDTO,
+            @RequestBody CommonDataNodeRegisterDTO commonDataNodeRegisterDTO,
             Principal principal
     ) throws PermissionDeniedException, NotExistException, AlreadyExistException, BindException, ValidationException {
 
-        validateOrganization(commonDataNodeRegisterDTO.getOrganization());
-
+        //for the first DN update all fields (name, description, organization) are mandatory in commonDataNodeRegisterDTO.
+        // In further updates they may be empty and will be taken from existing record
+        DN existingDN = baseDataNodeService.getById(dataNodeId);
+        if (existingDN.getName() == null) {
+            validateDNFields(commonDataNodeRegisterDTO);
+        }
         final IUser user = getUser(principal);
         final DN dataNode = conversionService.convert(commonDataNodeRegisterDTO, getDataNodeDNClass());
         dataNode.setId(dataNodeId);
-        final Organization organization = conversionService.convert(commonDataNodeRegisterDTO.getOrganization(), Organization.class);
+        final Organization organization;
+        if (commonDataNodeRegisterDTO.getOrganization() != null) {
+            organization = conversionService.convert(commonDataNodeRegisterDTO.getOrganization(), Organization.class);
+        } else {
+            organization = existingDN.getOrganization();
+        }
+        if (commonDataNodeRegisterDTO.getDescription() == null) {
```

Comment:
make checks consistently (in prev place you are comparing for not null, here for null). I'd suggest to pick `existingDn` and replace values in it (for newer ones) when it is applicable